### ISSUE TITLE
window.closeWhenEmpty - incorrect behavior? regression?

### ISSUE THREAD

**Issue by vraravam** [Microsoft/vscode#42898]
- VSCode Version: Version 1.20.0-insider (1.20.0-insider)
- OS Version: Mac OSX High Sierra version 10.13.3 (17D47)

Steps to Reproduce:

1. Open vscode from within a terminal window with the command `code-insiders .`
2. Since this is opening from my project directory, I assume it opens as the "Project Window"?
3. In my previous session, I had some open editor tabs. I closed each of them by using `cmd + w`
4. After closing the last editor, if I press `cmd + w` once more, the whole VS Code window is closed.
5. Expected behavior: if this setting is set to `false`, I expect that the window remains open even though there are no open editors.

Does this issue occur when all extensions are disabled?: Yes

Note: I had moved to the insiders build a few weeks ago, just since this issue was fixed in insiders, but had not yet been released in the stable version. As of the past 2 days, when I updated the insiders build, this has regressed.

**Comment by bpasero** [comment]
@vraravam that setting is about closing the window when you close the last editor, not about keeping the window open when you press `Cmd+W` after no editor is opened. But you can change your keybindings to not close the window when no editor is opened. 

Do you see a different behaviour in stable?

**Comment by vraravam** [comment]
@bpasero - tx for the clarification - but the way I described [my] intended/expected behavior was how the app was functioning till a couple of days ago.

> But you can change your keybindings to not close the window when no editor is opened.

Could you please tell me how to get this done?

**Comment by bpasero** [comment]
@vraravam by setting this in the keybindings for example:

{
	"key": "cmd+w",
	"command": "workbench.action.closeActiveEditor"
}

Can you please clarify what behaviour you saw previously that now is broken? For that, download VS Code stable (1.19.3), try your steps and compare to insiders.

**Comment by vraravam** [comment]
@bpasero - Thanks, the above key binding worked as expected.

As I stated in my original post, the stable works as the insiders ie `cmd+w` closes each active editor - and when no editors are open, it also closes the window. Also as mentioned, this was *not* the behavior when I switched from stable to insiders a few weeks ago.

**Comment by bpasero** [comment]
@vraravam the setting only works in a window that has no folder or workspace opened, could that be the difference?

**Comment by vraravam** [comment]
@bpasero - could be the case. I just switched over from atom to vs code. And though I usually only open the editor from the command line in the context of the current folder (which is the project that I am working on), there might have been some rare cases where I opened it from Spotlight - and while I was configuring the app preferences, I might have closed the project folder with a `cms+shift+w` and then proceeded. I see your point, and thanks for trying to help. My usecase is sorted with the overridden keybinding that you showed up above.
